bindings/rascal/utils/radial_basis.py

```diff
@@ -0,0 +1,415 @@
+"""
+A collection of utility functions to manipulate the radial basis.
+"""
+
+from scipy.special import legendre, gamma
+from copy import deepcopy
+import numpy as np
+from ..representations.spherical_expansion import SphericalExpansion
+
+
+def radial_basis_functions_dvr(
```

Comment:
Why is this function (and `radial_basis_functions_gto`) needed? It seems it is only used in an example to look at the radial function after optimization, right?

I would rather move the function to the example notebook then, rather than having it inside the rascal package. 

Otherwise, this is a duplicated implementation with code in C++, and we need to make sure they stay in sync. At the very least, there should be a big `WARNING` comment both here and in the C++ implementation to ensure they are kept in sync.

Reply:
as @ceriottm said, the GTO and DVR function are not implemented on the c++ side, only their expansion on Gaussian density function. The things we could use from our c++ implementation, are computed with numpy and scipy on the python e.g. the DVR grid with `np.polynomial.legendre.leggauss`, therefore do not produce code repetitions. There are tests checking the equality between numpy and c++ code in the scripts folder (e.g. see `generate_gauss_legendre_ref_data.py`)

Reply:
ok, that makes sense then. Could we modify the script producing the reference data using these new functions? This way, we should have an easier time keeping the definitions in sync!

Reply:
I am not sure about every function we use in the script but for 
```python
def radial_basis_functions_dvr(...): # <--- this function does not exist on c++ side
    # ...
    ... = np.polynomial.legendre.leggauss(...) # <--- this function exists on c++ side and is compared within a test
    # ....
```
Since there is not `radial_basis_functions_dvr` on the c++ side we cannot generate reference data for it. You don't need to compute the dvr basis functions for an expansion

Reply:
ok, fair enough :smiley: